src/ObjCRuntime/Runtime.cs

```diff
@@ -1310,19 +1337,61 @@ static void MissingCtor (IntPtr ptr, IntPtr klass, Type type, MissingCtorResolut
 			return ConstructNSObject<T> (ptr, typeof (T), MissingCtorResolution.ThrowConstructor1NotFound);
 		}
 
-		// The generic argument T is only used to cast the return value.
 		// The 'selector' and 'method' arguments are only used in error messages.
-		static T? ConstructNSObject<T> (IntPtr ptr, Type type, MissingCtorResolution missingCtorResolution) where T : class, INativeObject
+		static T? ConstructNSObject<T> (IntPtr ptr, Type type, MissingCtorResolution missingCtorResolution) where T : NSObject
 		{
 			return ConstructNSObject<T> (ptr, type, missingCtorResolution, IntPtr.Zero, default (RuntimeMethodHandle));
 		}
 
-		// The generic argument T is only used to cast the return value.
 		// The 'selector' and 'method' arguments are only used in error messages.
+#if NET
+		static T? ConstructNSObject<T> (IntPtr ptr, Type type, MissingCtorResolution missingCtorResolution, IntPtr sel, RuntimeMethodHandle method_handle) where T : NSObject, INSObjectFactory
+#else
 		static T? ConstructNSObject<T> (IntPtr ptr, Type type, MissingCtorResolution missingCtorResolution, IntPtr sel, RuntimeMethodHandle method_handle) where T : class, INativeObject
+#endif
 		{
 			if (type is null)
 				throw new ArgumentNullException (nameof (type));
+#if NET
+			if (Runtime.IsManagedStaticRegistrar) {
+				T? instance = default;
+				var nativeHandle = new NativeHandle (ptr);
+
+				// We want to create an instance of `type` and if we have the chance to use the factory method
+				// on the generic type, we will prefer it to using the lookup table.
+				if (typeof (T) == type // TODO can I drop the IsGenericType check? would it break some test?
```

Comment:
Could you resolve this TODO somehow?

Reply:
Thanks for pointing this out. I think the condition has changed since I wrote the comment and I just forgot to remove the comment. I'll double check that the comment is outdated and I'll get rid of it or address it.